scene/gui/foldable_container.cpp

```diff
@@ -0,0 +1,410 @@
+				c->set_visible(expanded);
+				fit_child_in_rect(c, Rect2(ofs, size));
+			}
+		} break;
+
+		case NOTIFICATION_MOUSE_EXIT: {
+			if (is_hovering) {
+				is_hovering = false;
+				queue_redraw();
+			}
+		} break;
+
+		case NOTIFICATION_LAYOUT_DIRECTION_CHANGED: {
+			set_title_alignment(title_alignment);
+		} break;
+
+		case NOTIFICATION_THEME_CHANGED: {
+			_shape();
+			update_minimum_size();
+			queue_redraw();
+		} break;
+	}
+}
+
+Size2 FoldableContainer::_get_title_panel_min_size() const {
+	Size2 s = Size2();
+	Ref<StyleBox> title_style = expanded ? theme_cache.title_style : theme_cache.title_collapsed_style;
+	Ref<Texture2D> icon = _get_title_icon();
+
+	if (title_style.is_valid()) {
+		s += title_style->get_minimum_size();
+	}
+	if (theme_cache.title_font.is_valid()) {
+		if (title.length() > 0) {
+			s.width += icon.is_valid() ? MAX(0, theme_cache.h_separation) : 0;
+			s += text_buf->get_size();
+		}
+	}
+	if (icon.is_valid()) {
+		s.width += icon->get_width();
+		if (title_style.is_valid()) {
+			s.height = MAX(s.height, title_style->get_minimum_size().height + icon->get_height());
+		} else {
+			s.height = MAX(s.height, icon->get_height());
+		}
+	}
+	return s;
+}
+
+Ref<StyleBox> FoldableContainer::_get_title_style() const {
+	if (is_hovering) {
+		return expanded ? theme_cache.title_hover_style : theme_cache.title_collapsed_hover_style;
+	}
+	return expanded ? theme_cache.title_style : theme_cache.title_collapsed_style;
+}
+
+Ref<Texture2D> FoldableContainer::_get_title_icon() const {
+	if (expanded) {
+		return theme_cache.arrow;
+	} else if (is_layout_rtl()) {
+		return theme_cache.arrow_collapsed_mirrored;
+	}
+	return theme_cache.arrow_collapsed;
+}
+
+void FoldableContainer::_shape() {
+	text_buf->clear();
+
+	if (!theme_cache.title_font.is_valid()) {
+		return;
+	}
+	text_buf->add_string(atr(title), theme_cache.title_font, theme_cache.title_font_size, language);
+}
+
+void FoldableContainer::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("set_expanded", "expanded"), &FoldableContainer::set_expanded);
+	ClassDB::bind_method(D_METHOD("is_expanded"), &FoldableContainer::is_expanded);
+	ClassDB::bind_method(D_METHOD("set_title", "title"), &FoldableContainer::set_title);
+	ClassDB::bind_method(D_METHOD("get_title"), &FoldableContainer::get_title);
+	ClassDB::bind_method(D_METHOD("set_title_alignment", "alignment"), &FoldableContainer::set_title_alignment);
+	ClassDB::bind_method(D_METHOD("get_title_alignment"), &FoldableContainer::get_title_alignment);
+	ClassDB::bind_method(D_METHOD("set_language", "language"), &FoldableContainer::set_language);
+	ClassDB::bind_method(D_METHOD("get_language"), &FoldableContainer::get_language);
+	ClassDB::bind_method(D_METHOD("set_text_direction", "text_direction"), &FoldableContainer::set_text_direction);
+	ClassDB::bind_method(D_METHOD("get_text_direction"), &FoldableContainer::get_text_direction);
+
+	ADD_SIGNAL(MethodInfo("expanded", PropertyInfo(Variant::BOOL, "is_expanded")));
```

Comment:
Can't name the signal and the property the same

Reply:
ok, I also was wondering about that error. Thank you all.